lib/ansible/module_utils/k8s/raw.py

```diff
@@ -143,7 +146,6 @@ def perform_action(self, resource, definition):
                         self.warn("{0} was not found, but creating it returned a 409 Conflict error. This can happen \
                                   if the resource you are creating does not directly create a resource of the same kind.".format(name))
                         return result
-                    result['result'] = k8s_obj.to_dict()
```

Comment:
It looks like this is only added in check mode now

Reply:
You're quite right - not quite sure how I stuffed that up, but glad you spotted it.

Fixing this properly also fixes your other comment.